core/heap.c

```diff
@@ -166,31 +166,11 @@ DECLARE_NEVERPROT_VAR(static bool out_of_vmheap_once, false);
 #define MEMSET_HEADER(p, value) VARIABLE_SIZE(p) = HEAP_TO_PTR_UINT(value)
 #define GET_VARIABLE_ALLOCATION_SIZE(p) (VARIABLE_SIZE(p) + HEADER_SIZE)
 
-/* heap is allocated in units
- * we start out with a small unit, then each additional unit we
- * need doubles in size, up to a maximum, we default to 32kb initial size
- * (24kb useful with guard pages), max size defaults to 64kb (56kb useful with
- * guard pages), we keep the max small to save memory, it doesn't seem to be
- * perf hit! Though with guard pages we are wasting quite a bit of reserved
- * (though not committed) space */
-/* the only big things global heap is used for are pc sampling
- * hash table and sideline sampling hash table -- if none of those
- * are in use, 16KB should be plenty, we default to 32kb since guard
- * pages are on by default (gives 24kb useful) max size is same as for
- * normal heap units.
- */
-/* the old defaults were 32kb (usable) for initial thread private and 16kb
- * (usable) for initial global, changed to simplify the logic for allocating
- * in multiples of the os allocation granularity.  The new defaults prob.
- * make more sense with the shared cache then the old did anyways.
- */
-/* restrictions -
- * any guard pages are included in the size, size must be > UNITOVERHEAD
- * for best performance sizes should be of the form
- * 2^n * page_size (where n is a positve integer) and max should be a multiple
```

Comment:
initial_heap_unit_size is set to 24 * 1024 in this PR. This old comment suggests that might not be optimal.

Reply:
initial_heap_unit_size is the same in this PR as it was before: 24K plus two guard pages if enabled.  The doubling behavior will work out b/c it will hit the max and take that value no matter what the doubling would result in.  But, we could shrink it to the power-of-2 16K, which might be better anyway: these days the per-thread units are rarely used (well, for -disable_traces anyway, and for many-threaded apps even with traces the traces are likely to be built by other threads) and should be kept small.  I tried to avoid changing the actual usable size in the PR to minimize changes: but maybe shrinking to 16K for more "natural" doubling is good.

Reply:
Given that the global and the nonpers init sizes are 24K too, I'd rather have fewer changes in this PR so I left this as 24K.

Reply:
Sure, makes sense to keep changes in each PR minimal. 

Maybe a TODO if you think the additional change might be valuable in future.

Reply:
> Maybe a TODO if you think the additional change might be valuable in future.

No, the 2^n under the max is not important and does not really help: better to save space.